packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -184,9 +211,39 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
     rtl: context.rtl,
   });
 
-  let grid = getDayGrid(gridOptions);
-  // Slicing because grid contains extra 1 week in the front and in the back.
-  grid = grid.slice(1, grid.length - 1);
+  const grid = React.useMemo(() => {
+    const grid = getDayGrid(gridOptions);
+    if (!showWeekNumbers) {
+      // Slicing because grid contains extra 1 week in the front and in the back.
+      return grid.slice(1, grid.length - 1);
+    }
+    return grid;
+  }, [gridOptions, showWeekNumbers]);
+
+  const renderHeader = () => {
+    return createShorthand(DatepickerCalendarHeader, header, {
+      defaultProps: () => ({
+        styles: resolvedStyles.header,
+        content: formatMonthYear(gridOptions.navigatedDate, localizedStrings),
+      }),
+      overrideProps: {
```

Comment:
This should be a function to handle original calls, see other `overrideProps` usages with callbacks for reference